File: mycuration/src/main/java/com/phicdy/mycuration/presentation/presenter/SelectFilterTargetRssPresenter.java
package com.phicdy.mycuration.presentation.presenter;

import android.support.annotation.NonNull;
import android.view.MenuItem;

import com.phicdy.mycuration.R;
import com.phicdy.mycuration.presentation.view.SelectTargetRssView;

public class SelectFilterTargetRssPresenter implements Presenter {

    private SelectTargetRssView view;

    public SelectFilterTargetRssPresenter() {
    }

    public void setView(@NonNull SelectTargetRssView view) {
        this.view = view;
    }

    @Override
    public void create() {
    }

    @Override
    public void resume() {
    }

    @Override
    public void pause() {

    }

    public void optionItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.done_select_target_rss:
                view.finishSelect();
        }
    }
}

File: legacy/src/main/java/com/phicdy/mycuration/presentation/presenter/SelectFilterTargetRssPresenter.kt
package com.phicdy.mycuration.presentation.presenter

import android.view.MenuItem
import com.phicdy.mycuration.legacy.R

import com.phicdy.mycuration.presentation.view.SelectTargetRssView

class SelectFilterTargetRssPresenter(private val view: SelectTargetRssView) : Presenter {

    override fun create() {}

    override fun resume() {}

    override fun pause() {}

    fun optionItemSelected(item: MenuItem) {
        when (item.itemId) {
            R.id.done_select_target_rss -> view.finishSelect()
        }
    }
}
